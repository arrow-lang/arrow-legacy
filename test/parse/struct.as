struct Empty { }
struct Box { value: int }
struct Point { x: int, y: int, }
