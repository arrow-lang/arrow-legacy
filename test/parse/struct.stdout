{
  "tag": "Module",
  "span": "test/parse/struct.as:1.1-3.7",
  "name": "struct",
  "block": {
    "tag": "Block",
    "span": "test/parse/struct.as:1.1-3.7",
    "statements": [
      {
        "tag": "Structure",
        "span": "test/parse/struct.as:1.1-7",
        "name": "Empty",
        "exported": false,
        "members": []
      },
      {
        "tag": "Structure",
        "span": "test/parse/struct.as:2.1-7",
        "name": "Box",
        "exported": false,
        "members": [
          {
            "tag": "Member",
            "span": "test/parse/struct.as:2.14-24",
            "name": "value",
            "type": {
              "tag": "TypePath",
              "span": "test/parse/struct.as:2.21-24",
              "segments": [
                "int"
              ]
            },
            "default_value": null
          }
        ]
      },
      {
        "tag": "Structure",
        "span": "test/parse/struct.as:3.1-7",
        "name": "Point",
        "exported": false,
        "members": [
          {
            "tag": "Member",
            "span": "test/parse/struct.as:3.16-22",
            "name": "x",
            "type": {
              "tag": "TypePath",
              "span": "test/parse/struct.as:3.19-22",
              "segments": [
                "int"
              ]
            },
            "default_value": null
          },
          {
            "tag": "Member",
            "span": "test/parse/struct.as:3.24-30",
            "name": "y",
            "type": {
              "tag": "TypePath",
              "span": "test/parse/struct.as:3.27-30",
              "segments": [
                "int"
              ]
            },
            "default_value": null
          }
        ]
      }
    ]
  }
}
